kolibri/plugins/learn/assets/src/views/LibraryPage.vue

```diff
@@ -296,26 +419,39 @@
             getParams.keywords = this.searchTerms.keywords;
           }
           ContentNodeResource.fetchCollection({ getParams }).then(data => {
-            this.results = data.results;
+            this.results = data.results.map(normalizeContentNode);
             this.more = data.more;
             this.labels = data.labels;
             this.searchLoading = false;
           });
         }
+        // else {
+        //   ContentNodeResource.fetchCollection({ getParams }).then(data => {
+        //     this.labels = data.labels;
+        //   });
+        // }
       },
       searchMore() {
         if (this.displayingSearchResults && this.more && !this.moreLoading) {
           this.moreLoading = true;
           ContentNodeResource.fetchCollection({ getParams: this.more }).then(data => {
-            this.results.push(...data.results);
+            this.results.push(...data.results.map(normalizeContentNode));
             this.more = data.more;
             this.labels = data.labels;
             this.moreLoading = false;
           });
         }
       },
-      toggleInfoPanel(content) {
-        this.sidePanelContent = content;
+      removeFilterTag(value) {
+        let query = JSON.parse(JSON.stringify(this.$route.query));
+        const key = Object.keys(query).filter(function(key) {
+          return query[key] === value;
+        })[0];
+        delete query[key];
+        this.$router.replace({ query: query });
+      },
+      clearSearch() {
+        this.$router.push(this.$router.getRoute(PageNames.LIBRARY));
```

Comment:
```suggestion
        this.searchTerms = {};
```

Would also work here.